python_otbr_api/__init__.py

```diff
@@ -0,0 +1,53 @@
+"""API to interact with the Open Thread Border Router REST API."""
+from __future__ import annotations
+from http import HTTPStatus
+
+import aiohttp
+
+
+class OTBRError(Exception):
+    """Raised on error."""
+
+
+class OTBR:
+    """Class to interact with the Open Thread Border Router REST API."""
+
+    def __init__(
+        self, url: str, session: aiohttp.ClientSession, timeout: int = 10
+    ) -> None:
+        """Initialize."""
+        self._session = session
+        self._url = url
+        self._timeout = timeout
+
+    def _raise_for_status(self, response: aiohttp.ClientResponse) -> None:
+        """Raise if status >= 400."""
+        try:
+            response.raise_for_status()
+        except aiohttp.ClientResponseError as exc:
+            raise OTBRError(f"unexpected http status {response.status}") from exc
```

Comment:
This will have us lose any error body. Shouldn’t we just have it raise aiohttp errors?

Reply:
Fixed